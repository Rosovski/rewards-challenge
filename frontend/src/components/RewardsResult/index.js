import useFetch from "../../utils/hooks/useFetch";

const RewardsResult = ({ customerId }) => {
  const rewardsData = useFetch(`/api/rewards/${customerId}`);

  return (
    <div data-testid="rewards-component">
      {rewardsData !== null && (
        <h2 data-testid="rewards-result">
          The rewards that user earned in past 3 months is: {rewardsData} points
        </h2>
      )}
    </div>
  );
};

export default RewardsResult;
