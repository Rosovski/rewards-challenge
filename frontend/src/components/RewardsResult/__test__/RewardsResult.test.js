import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import RewardsResult from "..";
import apiClient from "../../../utils/apiClient";

afterEach(cleanup);

jest.mock("../../../utils/apiClient");

describe("test suite for the component of rewards calculation result", () => {
  it("render the component", () => {
    render(<RewardsResult customerId="1" />);
  });

  it("mock the api call to get response", async () => {
    apiClient.get.mockImplementationOnce(() =>
      Promise.resolve({ data: { result: 420 } })
    );

    const { findByText, findByTestId } = render(
      <RewardsResult customerId="1" />
    );

    const result = await findByText(/420/);
    const resultDisplay = await findByTestId("rewards-result");

    expect(result).toBeInTheDocument();
    expect(resultDisplay).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<RewardsResult customerId="1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
