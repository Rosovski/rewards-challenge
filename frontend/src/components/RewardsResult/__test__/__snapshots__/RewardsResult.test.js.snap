// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for the component of rewards calculation result matches snapshot 1`] = `
<div
  data-testid="rewards-component"
>
  <h2
    data-testid="rewards-result"
  >
    The rewards that user earned in past 3 months is: 
     points
  </h2>
</div>
`;
